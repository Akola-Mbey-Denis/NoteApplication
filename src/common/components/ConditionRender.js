import React, { Component }from 'react';

// const ConditionRender = (renderIf,props) => {
//   return function (renderIf) {
//     if (!renderIf) {
//       return null;
//     }
//     return <Component {...props} />;
//   };
// };

export const ConditionRenderComponent= ({
  renderIf,
  children
}) => {
  if (!renderIf) {
    return null;
  }
  return <>{children}</>;
};

// export default ConditionRender;
