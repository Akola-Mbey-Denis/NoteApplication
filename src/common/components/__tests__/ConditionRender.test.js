import React from 'react';
import {render} from '@testing-library/react-native';
import  {ConditionRenderComponent} from '../ConditionRender';
import {Text} from 'react-native';

const TestComponent = (props) => {
  return <Text>Some content to be rendered</Text>;
};

// describe('ConditionRender', () => {
//   it('should render component', () => {
//     const Component = ConditionRender(TestComponent);
//     const screen = render(<Component renderIf />);

//     expect(screen.toJSON()).toMatchSnapshot();
//     expect(screen.getByText(/Some content to be rendered/)).toBeDefined();
//   });

//   it('should not render component', () => {
//     const Component = ConditionRender(TestComponent);
//     const screen = render(<Component renderIf={false} />);

//     expect(screen.toJSON()).toMatchSnapshot();
//     expect(screen.queryByText(/Some content to be rendered/)).toBeNull();
//   });
// });

describe('ConditionRenderComponent', () => {
  it('should render component', () => {
    const screen = render(
      <ConditionRenderComponent renderIf>
        <TestComponent />
      </ConditionRenderComponent>,
    );

    expect(screen.toJSON()).toMatchSnapshot();
    expect(screen.getByText(/Some content to be rendered/)).toBeDefined();
  });

  it('should not render component', () => {
    const screen = render(
      <ConditionRenderComponent renderIf={false}>
        <TestComponent />
      </ConditionRenderComponent>,
    );

    expect(screen.toJSON()).toMatchSnapshot();
    expect(screen.queryByText(/Some content to be rendered/)).toBeNull();
  });
});
