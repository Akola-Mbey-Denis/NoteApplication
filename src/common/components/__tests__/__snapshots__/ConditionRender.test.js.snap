// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionRenderComponent should not render component 1`] = `null`;

exports[`ConditionRenderComponent should render component 1`] = `
<Text>
  Some content to be rendered
</Text>
`;
